src/errors.rs

```diff
@@ -1,74 +1,102 @@
     #[error("error while sending payload: [{0}]")]
     PayloadSendError(reqwest::Error),
 }
 
+/// Errors returned when the client has an issue parsing the response of a method call.
 #[derive(Debug, Error)]
 pub enum JsonRpcTransportHandlerResponseError {
+    /// An error returned when the client is unable to parse the result of a method call.
     #[error("error while parsing method call result: [{0}]")]
     ResultParseError(serde_json::Error),
+    /// An error returned when the client is unable to parse the error message returned when a problematic method call is made.
     #[error("error while parsing method call error message: [{0}]")]
     ErrorMessageParseError(serde_json::Error),
 }
 
+/// Errors returned while receiving responses from an rpc server.
 #[derive(Debug, Error)]
 pub enum JsonRpcTransportRecvError {
+    /// An error returned when the receives an error that does not fit any of the known error types.
     #[error("unexpected server response: [{0:?}]")]
     UnexpectedServerResponse(Message),
+    /// An error returned when the client is unable to read the response from the rpc server.
     #[error("error while reading response: [{0}]")]
     PayloadRecvError(reqwest::Error),
+    /// An error returned when the base response structure is malformed e.g. meta properties like rpc version are missing.
     #[error("error while parsing server response: [{0:?}]")]
     PayloadParseError(message::Broken),
+    /// An error returned when the internal response structure is malformed, e.g. block data in a block method response is missing.
     #[error(transparent)]
     ResponseParseError(JsonRpcTransportHandlerResponseError),
 }
 
+/// Errors returned while sending requests to or receiving responses from the rpc server.
 #[derive(Debug, Error)]
 pub enum RpcTransportError {
+    /// Errors returned while sending a request to the rpc node.
     #[error(transparent)]
     SendError(JsonRpcTransportSendError),
+    /// Errors returned while receiving a response from an rpc server.
     #[error(transparent)]
     RecvError(JsonRpcTransportRecvError),
 }
 
+/// Meta errors returned by the rpc server.
 #[derive(Debug, Error)]
 pub enum JsonRpcServerResponseStatusError {
+    /// An error returned when the rpc client is unauthorized.
     #[error("this client is unauthorized")]
     Unauthorized,
+    /// An error returned when the rpc client exceeds the rate limit by sending too many requests.
     #[error("this client has exceeded the rate limit")]
     TooManyRequests,
+    /// An error returned when the rpc server returns a non-200 status code.
     #[error("the server returned a non-OK (200) status code: [{status}]")]
     Unexpected { status: reqwest::StatusCode },
 }
 
+/// Errors returned by the rpc server.
 #[derive(Debug, Error)]
 pub enum JsonRpcServerError<E> {
+    /// An error returned when the rpc server is unable to validation a client's request.
```

Comment:
This is returned exclusively for Invalid RPC methods. Technically, we should never have this issue with the client (if you stay away from `methods::any`). Still worth documenting, nonetheless.

Reply:
Okay, so this isn't just for invalid RPC methods. But also for cases where the RPC method is unable to parse the provided arguments.

Reply:
Or if the payload sent to the server, while a valid JSON RPC message, isn't structured as a request.

The inverse of https://github.com/near/near-jsonrpc-client-rs/blob/091fb5385d0212716f343e0cbf7752ea1ce210c4/src/errors.rs#L34-L36